File: mapreduce/patent-java/src/main/java/com/epam/bigdata/mapreduce/patent/PatentCitationMapper.java
package com.epam.bigdata.mapreduce.patent;

import org.apache.hadoop.io.Text;
import org.apache.hadoop.mapreduce.Mapper;

import java.io.IOException;

public class PatentCitationMapper extends Mapper<Text, Text, Text, Text> {
    @Override
    protected void map(Text key, Text value, Context context) throws IOException, InterruptedException {
        String[] citation = key.toString().split(",");
        Text cited = new Text(citation[0]);
        Text citing = new Text(citation[1]);
        context.write(cited, citing);
    }
}

File: mapreduce/patent-scala/src/main/scala/com/epam/bigdata/mapreduce/patent/PatentCitationMapper.scala
package com.epam.bigdata.mapreduce.patent

import org.apache.hadoop.io.Text
import org.apache.hadoop.mapreduce.Mapper

class PatentCitationMapper extends Mapper[Text, Text, Text, Text] {
   override def map(key: Text, value: Text, context: Mapper[Text, Text, Text, Text]#Context) = {
      val write = context.write _
      write.tupled(key.toString split (",") match { case Array(a, b) => (new Text(a), new Text(b)) })
   }
}
